torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -196,6 +208,34 @@ class TensorExprFuser {
       updateQueue(fusion_group, queue, visited_nodes);
     }
 
+    // Update typeinfo for inputs and outputs of the fusion group
```

Comment:
nitpick: can we refactor this into a helper as this blob of logic has a very specific purpose.